play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSRequest.scala

```diff
@@ -65,18 +67,31 @@ case class StandaloneAhcWSRequest(
   override def withAuth(username: String, password: String, scheme: WSAuthScheme): Self =
     copy(auth = Some((username, password, scheme)))
 
-  override def withHeaders(hdrs: (String, String)*): Self = {
-    val headers = hdrs.foldLeft(this.headers)((m, hdr) =>
-      if (m.contains(hdr._1)) m.updated(hdr._1, m(hdr._1) :+ hdr._2)
-      else m + (hdr._1 -> Seq(hdr._2)))
-    copy(headers = headers)
+  override def setHeaders(hdrs: (String, String)*): Self = {
+    var newHeaders = hdrs.foldLeft(TreeMap[String, Seq[String]]()(CaseInsensitiveOrdered)) {
```

Comment:
See #48. I think we should switch to a list for both query params and headers. The order does actually matter for some headers with the same name, and this will fix that bug.